plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveUpdatablePageSource.java

```diff
@@ -116,28 +117,31 @@ public HiveUpdatablePageSource(
     @Override
     public void deleteRows(Block rowIds)
     {
-        List<Block> blocks = rowIds.getChildren();
-        checkArgument(blocks.size() == 3, "The rowId block for DELETE should have 3 children, but has %s", blocks.size());
```

Comment:
Should this check survive in the form of `columnarRow.getFieldCount()==3` inside `deleteRowsInternal`?

Reply:
Good idea, added.